server/services/v1/query_runner.go

```diff
@@ -720,6 +730,96 @@ func (runner *SearchQueryRunner) getFieldSelection(collFields []*schema.Queryabl
 	return factory, nil
 }
 
+type SubscribeQueryRunner struct {
+	*BaseQueryRunner
+
+	req       *api.SubscribeRequest
+	streaming SubscribeStreaming
+}
+
+func (runner *SubscribeQueryRunner) Run(ctx context.Context, tx transaction.Tx, tenant *metadata.Tenant) (*Response, context.Context, error) {
+	db, err := runner.GetDatabase(ctx, tx, tenant, runner.req.GetDb())
+	if ulog.E(err) {
+		return nil, ctx, err
+	}
+
+	collection, err := runner.GetCollections(db, runner.req.GetCollection())
+	if ulog.E(err) {
+		return nil, ctx, err
+	}
+
+	table, err := runner.encoder.EncodeTableName(tenant.GetNamespace(), db, collection)
+	if ulog.E(err) {
+		return nil, ctx, err
+	}
+
+	skipFirst := false
+	startTime := time.Now().UTC().Format(time.RFC3339Nano)
+
+	ticker := time.NewTicker(100 * time.Millisecond)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		tickerTx, err := runner.txMgr.StartTx(ctx)
+		if ulog.E(err) {
+			return nil, ctx, err
+		}
+
+		startKey, err := runner.encoder.EncodeKey(table, collection.Indexes.PrimaryKey, []interface{}{startTime})
+		if ulog.E(err) {
+			return nil, ctx, err
+		}
+
+		end, err := time.Parse(time.RFC3339, startTime)
+		if ulog.E(err) {
+			return nil, ctx, err
+		}
+
+		endTime := end.Add(34 * time.Second).Format(time.RFC3339)
```

Comment:
We still need to do a snapshot read otherwise the writes may see conflict in this window but that can be done in next diff.

Reply:
Yeah, will do next iteration.